lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -870,6 +870,16 @@ llvm::json::Value CreateThread(lldb::SBThread &thread, lldb::SBFormat &format) {
   return llvm::json::Value(std::move(object));
 }
 
+llvm::json::Array GetThreads(lldb::SBProcess process, lldb::SBFormat &format) {
+  llvm::json::Array threads;
+  const uint32_t num_threads = process.GetNumThreads();
+  for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
+    lldb::SBThread thread = process.GetThreadAtIndex(thread_idx);
+    threads.emplace_back(CreateThread(thread, format));
+  }
+  return threads;
```

Comment:
This should take the TargetAPI lock as it's doing two SB API calls consecutively. 
```suggestion
  lldb::SBMutex lock = process.GetTarget().GetAPIMutex();
  std::lock_guard<lldb::SBMutex> guard(lock);
  
  llvm::json::Array threads;
  const uint32_t num_threads = process.GetNumThreads();
  for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
    lldb::SBThread thread = process.GetThreadAtIndex(thread_idx);
    threads.emplace_back(CreateThread(thread, format));
  }
  return threads;
```

Reply:
Might be nice to add a `lldb::SBThreadList` to the API and then add:
```
lldb::SBThreadList lldb::SBProcess::GetThreads();
```

Reply:
(we don't need to add the API for this patch, I was just mentioning this). The API lock fix will work here nicely.